File: LeetCode/0066_PlusOne/Solution.java
public class Solution {
    public int[] plusOne(int[] digits) {
    	int n = digits.length;
        for(int i = n - 1; i>=0; i--){
        	if(digits[i]<9){
        		digits[i]++;
        		return digits;
        	}

        	digits[i] = 0;
        }

        int[] newDigit = new int[n+1];
        newDigit[0] = 1;

        return newDigit;
    }
}

/*
The last part of code is only for the case that the whole input array is 9s.
For example : 99999-----> 100000
Any other case would return in the loop.
*/

File: LeetCode/0066_PlusOne/Solution.py
import math
import functools

class Solution:
    # def plusOne(self, digits: List[int]) -> List[int]:
    def plusOne1(self, digits):
        num = 0
        for i in range(len(digits)):
            num = num*10 + digits[i]
        return [int(i) for i in str(num + 1)]

    def plusOne(self, digits):
        num = functools.reduce(lambda x, y: x * 10 + y, digits) + 1
        return [int(i) for i in str(num)]

